ark/api/v1alpha1/query_types.go

```
@@ -118,6 +145,73 @@ type QueryList struct {
 	Items           []Query `json:"items"`
 }
 
+// GetInputString returns the input as a string when type="user" or type is empty (default)
+func (q *QuerySpec) GetInputString() (string, error) {
+	if q.Type != "" && q.Type != QueryTypeUser {
+		return "", fmt.Errorf("cannot get string input for type=%s, expected type=%s or empty", q.Type, QueryTypeUser)
+	}
+
+	var inputString string
+	if err := json.Unmarshal(q.Input.Raw, &inputString); err != nil {
+		return "", fmt.Errorf("failed to unmarshal input as string: %w", err)
+	}
+
+	return inputString, nil
+}
+
+// GetInputMessages returns the input as []Message when type="messages"
+func (q *QuerySpec) GetInputMessages() ([]Message, error) {
+	if q.Type != QueryTypeMessages {
+		return nil, fmt.Errorf("cannot get message input for type=%s, expected type=%s", q.Type, QueryTypeMessages)
+	}
+
+	var messages []Message
+	if err := json.Unmarshal(q.Input.Raw, &messages); err != nil {
```

what happens if the user uses other valid stuctures such as tool call messages? do we preserve them and send them on? the spec says we should forward all of the content [and this will be a *huge* help cause upcoming or model specific parameters are then supproted out of the box)

---

Great point. currently other fields are silently ignored. Do we want to address this now? or in a follow up? What should the Message structure be?